### ISSUE TITLE
WebSocket connection sends data after user has logged off

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-ui#478]
**Describe the bug**

@dwsutherland reviewed https://github.com/cylc/cylc-uiserver/pull/124#pullrequestreview-457164367 and his comment raised an interesting point: what happens to the WebSocket connection after the user logs off?

The answer is that the WebSocket connection keeps sending data. Tested on both `master` and on that PR's branch. New connections will be blocked after that PR is merged. But the existing connection is never terminated.

**Release version(s) and/or repository branch(es) affected?**

`master` (and also using the UIS PR for websocket auth decorator).

**Steps to reproduce the bug**

Start Cylc 8 env, open the browser console, look under Network for WS connections. Leave it open. In another tab log out of JupyterHub.

**Expected behavior**

The messages are supposed to stop, probably a connection terminated or broken connection error for the WebSocket too.

**Screenshots**

![image](https://user-images.githubusercontent.com/304786/88866268-44211900-d25e-11ea-9340-f7a23475429c.png)

**Additional context**

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by kinow** [comment]
The UI Server keeps running too. The `singleuser-app` (Notebook app running with JupyterHub) has the same behaviour.

The Notebook app also has a WebSocket connection. After you log out, you can still evaluate cells too, and the WebSocket gets the data. There's a "Forbidden" message added to the UI, but that doesn't stop requests.

![image](https://user-images.githubusercontent.com/304786/88871248-856bf580-d26b-11ea-8fdf-57ea19ee3fb1.png)

**Comment by dwsutherland** [comment]
Is this more a UIServer issue, right? i.e. the UIS needs to close the connection after logout?

**Comment by kinow** [comment]
Yeah, let me transfer that to the UIS @dwsutherland . And there is also a related issue in JupyterHub (which looks like I linked in another issue already, even though I have no recollection of doing so :rofl: )

https://github.com/jupyterhub/jupyterhub/issues/1780
